innovation_sweet_spots/pipeline/pilot/investment_predictions/create_dataset.py

```diff
@@ -0,0 +1,167 @@
+        # Add flags for company founded/acquired/went public in time window
+        .pipe(
+            utils.window_flag,
+            start_date=window_start_date,
+            end_date=window_end_date,
+            variable="founded_on",
+        )
+        .pipe(
+            utils.window_flag,
+            start_date=window_start_date,
+            end_date=window_end_date,
+            variable="acquired_on",
+        )
+        .pipe(
+            utils.window_flag,
+            start_date=window_start_date,
+            end_date=window_end_date,
+            variable="went_public_on",
+        )
+        # Filter companies founded in the time window but not acquired or went public
+        .query(
+            "founded_on_in_window == 1 & acquired_on_in_window == 0 & went_public_on_in_window == 0"
+        )
+        .drop(
+            columns=[
+                "founded_on_in_window",
+                "acquired_on_in_window",
+                "went_public_on_in_window",
+            ]
+        )
+        # Add flags for each measure of success
+        .pipe(
+            utils.future_flag,
+            start_date=success_start_date,
+            variable="funding_round_date",
+        )
+        .pipe(utils.future_flag, start_date=success_start_date, variable="acquired_on")
+        .pipe(
+            utils.future_flag, start_date=success_start_date, variable="went_public_on"
+        )
+        # Create future_success variable which is set to 1 if one of the above flags is 1
+        .assign(
+            future_success=lambda x: x[
+                [
+                    "future_funding_round_date",
+                    "future_acquired_on",
+                    "future_went_public_on",
+                ]
+            ].max(axis=1)
+        )
+        .drop(
+            columns=[
+                "future_funding_round_date",
+                "future_acquired_on",
+                "future_went_public_on",
+            ]
+        )
+        .pipe(
+            utils.add_n_funding_rounds_in_window,
+            start_date=window_start_date,
+            end_date=window_end_date,
+        )
+        .pipe(
+            utils.add_n_month_since_last_investment_in_window,
+            start_date=window_start_date,
+            end_date=window_end_date,
+        )
+        .pipe(utils.add_n_month_since_founded, end_date=window_end_date)
+        # Drop columns
+        .pipe(
+            utils.drop_multi_cols,
+            cols_to_drop_str_containing=[
+                "funding_round_date",
+                "funding_round_id",
+                "acquired_on",
+                "went_public_on",
+            ],
+        )
+        .drop(columns=["founded_on", "closed_on"])
+        .reset_index(drop=True)
+        # Save to csv
+        .to_csv(
+            pilot_outputs
+            / f"investment_predictions/company_data_window_{str(window_start_date).split(' ')[0]}-{str(window_end_date).split(' ')[0]}.csv"
+        )
+    )
```

Comment:
Wow, that's the longest panda chain I've ever seen! 🥇

Reply:
Ha, thanks! I think it might be too long and will get longer when more features are added. Might be worth combining some of the functions together later on.